File: server/routes/historyRouter.ts
import * as express from "express";
import {handleRequest} from "./route-helper";

class HistoryRoutes {
    list = handleRequest('performPairAssignmentDocumentListQuery');
    savePairs = handleRequest('performSavePairAssignmentDocumentCommand');
    deleteMember = handleRequest('performDeletePairAssignmentDocumentCommand');
}

const history = new HistoryRoutes();
const router = express.Router({mergeParams: true});
router.route('')
    .get(history.list)
    .post(history.savePairs);
router.delete('/:id', history.deleteMember);

export default router;

File: server/src/main/kotlin/com/zegreatrob/coupling/server/route/historyRouter.kt
package com.zegreatrob.coupling.server.route

import com.zegreatrob.coupling.server.external.express.Router

val historyRouter = Router(routerParams(mergeParams = true)).apply {
    route("")
        .post(dispatch { performSavePairAssignmentDocumentCommand })
    route("/:id")
        .delete(dispatch { performDeletePairAssignmentDocumentCommand })
}